adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -275,7 +275,17 @@ private AuthenticationResult tryAcquireTokenSilent(final AuthenticationRequest a
 
         if (shouldTrySilentFlow(authenticationRequest)) {
             Logger.v(TAG, "Try to acquire token silently, return valid AT or use RT in the cache.");
-            authenticationResult = acquireTokenSilentFlow(authenticationRequest);
+            try {
```

Comment:
When BrokerAccountService is binded and we call acquireToken for an account which doesn't exist in the broker. Adal will try going to look up the account in broker before trying interactive auth. So BrokerAccountService.acquireTokenSilently will be called to get the token in broker. createErrorBundle(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED) will be returned to BrokerAccountServiceHandler and invoke the onError() and AuthenticationException(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED) will be thrown to the calling method BrokerProxy.getAuthTokenInBackground(). And the exception will popped up all way to tryAcquireTokenSilent() and this is where caused a problem. If we don't catch the exception here, it will throw and reach to the callback.onError in the app before we trying the interavtive auth. We only want to throw the ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED error when it is a silent call.